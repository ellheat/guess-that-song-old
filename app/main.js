/**
 * app.js
 *
 * This is the entry file for the application, only setup and boilerplate
 * code.
 */

import React from 'react';
import ReactDOM from 'react-dom';

// Needed for redux-saga es6 generator support
import 'babel-polyfill';

// Import all the third party stuff
import { Provider } from 'react-redux';
import createHistory from 'history/createBrowserHistory';
import { ConnectedRouter } from 'react-router-redux';
import injectTapEventPlugin from 'react-tap-event-plugin';
import FontFaceObserver from 'fontfaceobserver';
import 'normalize.css/normalize.css';
import configureStore from './modules/store';
import UnsupportedBrowserDetection from './utils/unsupportedBrowserDetection';

// Observe loading of Open Sans (to remove open sans, remove the <link> tag in
// the index.html file and this observer)
const openSansObserver = new FontFaceObserver('Open Sans', {});

// When Open Sans is loaded, add a font-family using Open Sans to the body
openSansObserver.load().then(() => {
  document.body.classList.add('fontLoaded');
}, () => {
  document.body.classList.remove('fontLoaded');
});

// Needed for onTouchTap
// Check this repo:
// https://github.com/zilverline/react-tap-event-plugin
injectTapEventPlugin();

// Create redux store with history
// this uses the singleton browserHistory provided by react-router
// Optionally, this could be changed to leverage a created history
// e.g. `const browserHistory = useRouterHistory(createBrowserHistory)();`
const initialState = {};
const browserHistory = createHistory();
const store = configureStore(initialState, browserHistory);

if (process.env.NODE_ENV === 'development') {
  if (!window.__REDUX_DEVTOOLS_EXTENSION__) {
    const DevToolsComponent = require('./utils/devtools.component').default;
    const devToolsRoot = window.document.createElement('div');

    window.document.body.appendChild(devToolsRoot);

    ReactDOM.render(
      <Provider store={store}>
        <DevToolsComponent />
      </Provider>,
      devToolsRoot
    );
  }
}

const render = () => {
  const NextApp = require('./routes').default;

  ReactDOM.render(
    <Provider store={store}>
      <ConnectedRouter history={browserHistory}>
        <NextApp />
      </ConnectedRouter>
    </Provider>,
    document.getElementById('app')
  );
};

const initApp = () => {
  const detection = new UnsupportedBrowserDetection();
  if (!detection.isSupported()) {
    return;
  }

  // Chunked polyfill for browsers without Intl support
  if (!window.Intl) {
    (new Promise((resolve) => {
      resolve(require('intl'));
    }))
      .then(() => Promise.all([
        require('intl/locale-data/jsonp/en.js'),
      ]))
      .then(() => render())
      .catch((err) => {
        throw err;
      });
  } else {
    render();
  }
};

initApp();

// Install ServiceWorker and AppCache in the end since
// it's not most important operation and if main code fails,
// we do not want it installed
if (process.env.NODE_ENV === 'production') {
  require('offline-plugin/runtime').install(); // eslint-disable-line global-require, import/no-extraneous-dependencies
}
