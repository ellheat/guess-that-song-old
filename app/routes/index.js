import React, { Component } from 'react';
import { Route, Switch, Redirect, withRouter } from 'react-router-dom';
import PropTypes from 'prop-types';
import { DEFAULT_LOCALE } from '../modules/locales/locales.redux';

import App from './app.container';
import { Home } from './home';
import { NotFound } from './notFound';

export class RootContainer extends Component {
  static propTypes = {
    location: PropTypes.object.isRequired,
  };

  render() {
    return (
      <Switch>
        <Route exact path="/" render={() => <Redirect to={DEFAULT_LOCALE} />} />


        <Route exact path="/404" component={NotFound} />

        <Route path="/:lang">
          <App location={this.props.location}>
            <Switch>
              <Route exact path="/:lang" component={Home} />

              <Route component={NotFound} />
            </Switch>
          </App>
        </Route>
      </Switch>
    );
  }
}

export default withRouter(RootContainer);
