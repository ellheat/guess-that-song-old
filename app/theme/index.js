export { default as SpriteImage } from './spriteImage.component';

export * as sprites from './sprites';
