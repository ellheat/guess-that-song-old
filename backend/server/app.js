const serverless = require('serverless-http');
const express = require('express');
const path = require('path');
const port = require('../config/ports');
const app = express();
const router = express.Router(); // eslint-disable-line

const { getCharacters } = require('../config/characters');
const { getSpotifyToken, fetchPlaylist } = require('../utils/spotify');


const configureServer = () => new Promise((resolve) => {
  router.get('/', (req, res) => {
    res.send(express.static(path.join(__dirname, '../../build')));
  });
  app.listen(port.api, () => {
    console.log(`Backend listening on port ${port.api}!`.success); // eslint-disable-line
    getCharacters().then(() => console.log('Characters created'.success)); // eslint-disable-line
    getSpotifyToken().then(() => {
      console.log('Spotify token created'.success); // eslint-disable-line
      fetchPlaylist().then(() => resolve());
    });
  });
});


module.exports = {
  app,
  port,
  configureServer,
};

exports.handler = serverless(app);
