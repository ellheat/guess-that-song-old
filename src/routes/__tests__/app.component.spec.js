import React from 'react';
import { shallow } from 'enzyme';

import { App } from '../app.component';


describe('App: Component', () => {
  const children = <div className="app__children">Children</div>;
  const defaultProps = {
    language: 'en',
    location: {},
    getMovieStatus: () => {},
  };

  const component = (props) => (
    <App {...defaultProps} {...props}>
      {children}
    </App>
  );

  it('should render App when language is set', () => {
    const wrapper = shallow(component({ language: 'en' }));
    global.expect(wrapper).toMatchSnapshot();
  });
});
