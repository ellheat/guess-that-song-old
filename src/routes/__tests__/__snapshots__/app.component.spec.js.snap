// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App: Component should render App when language is set 1`] = `
<Fragment>
  <HelmetWrapper
    defaultTitle="MyRoma"
    defer={true}
    encodeSpecialCharacters={true}
    titleTemplate="%s - MyRoma"
  />
  <IntlProvider
    locale="en"
    location={Object {}}
    messages={
      Object {
        "cameraStream.imagePlaceholder": "Unfortunately, you have not agreed to share the camera. You can upload photo by using left button.",
        "grid.cta": "New message",
        "home.welcome": "Hello World!",
        "intro.button": "Explore MyRoma",
        "intro.paragraph": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pellentesque facilisis justo non convallis ornare.",
        "intro.title": "Alfonso, Mexico",
        "language.differentLanguage": "EspaÃ±ol",
        "language.differentLanguageShort": "ES",
        "menu.home": "Home",
        "menu.itemPlaceholder": "Lorem ipsum",
        "menu.privacy": "Privacy",
        "menu.terms": "Terms & conditions",
        "menu.trailer": "Trailer",
        "menu.watchFilm": "Watch film",
        "notFound.button": "Explore MyRoma",
        "notFound.firstParagraph": "We are very sorry for the inconvience.",
        "notFound.pageTitle": "Not found",
        "notFound.secondParagraph": "It looks like you are trying to access a page that do not exist.",
        "notFound.title": "Error: 404",
        "photo.photoFormatError": "Your file must be in a .jpg or .png format",
        "photo.placeholder": "Take a photo or upload an image",
        "photoForm.back": "Back",
        "photoForm.photoFormatError": "Your file must be in a .jpg or .png format",
        "photoForm.submit": "Send your message",
        "photoForm.textLabel": "Text",
        "photoForm.textMaxLengthError": "Your message should not be longer than 255 characters",
        "photoForm.textMinLengthError": "Your message should not be shorter than 2 characters",
        "photoForm.textPlaceholder": "Text",
        "photoForm.textRequiredError": "Either text or photo is required",
        "photoForm.userCountryLabel": "Country",
        "photoForm.userCountryPlaceholder": "Country",
        "photoForm.userCountryRequiredError": "Country is required",
        "photoForm.userNameLabel": "Name",
        "photoForm.userNameMaxLengthError": "The name should not be longer than 16 characters",
        "photoForm.userNameMinLengthError": "The name should not be shorter than 2 characters",
        "photoForm.userNamePlaceholder": "Name",
        "photoForm.userNameRequiredError": "Name is required",
        "photoSuccess.explore": "Explore messages",
        "photoSuccess.title": "Your message has been sent. The message will appear after moderator accepts it",
        "privacy.firstParagraph": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pellentesque facilisis justo non convallis ornare. Donec commodo rhoncus nunc quis rutrum.",
        "privacy.pageTitle": "Privacy",
        "privacy.secondParagraph": "Aliquam eu quam ultricies, consectetur ante at, rutrum tellus. Cras arcu nunc, lacinia eu faucibus quis, commodo eu risus. Fusce nibh turpis, mollis eu pharetra tincidunt, iaculis in est.",
        "privacy.thirdParagraph": "Vestibulum molestie orci libero, eget ullamcorper sapien cursus sit amet. Donec nec diam scelerisque libero imperdiet fermentum ac vitae lacus. In nibh dolor, sollicitudin ut purus maximus, dictum tincidunt magna. Quisque luctus dignissim erat et cursus. Nullam suscipit metus sed quam fermentum fermentum.",
        "privacy.title": "Privacy",
        "selectField.noOptionsMessage": "Not found",
        "share.title": "Share your message with the world:",
        "terms.firstParagraph": "1. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pellentesque facilisis justo non convallis ornare.",
        "terms.pageTitle": "Terms & Conditions",
        "terms.secondParagraph": "2. Donec commodo rhoncus nunc quis rutrum.",
        "terms.thirdParagraph": "3. Aliquam eu quam ultricies, consectetur ante at, rutrum tellus.",
        "terms.title": "Terms & Conditions",
        "trailer.description": "Time and Space constrain us, but they also define who we are, creating inexplicable bonds with the others that flow with us at the same time and through the same places.",
        "trailer.pageTitle": "Trailer",
        "trailer.title": "Roma Official Trailer",
      }
    }
  >
    <Connect(InjectIntl(withRouter(Header))) />
    <div
      className="app__children"
    >
      Children
    </div>
    <Connect(withRouter(Modal)) />
  </IntlProvider>
</Fragment>
`;
